### PR TITLE
chore: remove using local storage for persisting console settings on the frontend

### PR DESCRIPTION
**Pull request by hoxyq**
After https://github.com/facebook/react/pull/30636 and https://github.com/facebook/react/pull/30986 we no longer store settings on the Frontend side via `localStorage`.

This PR removes all occurrences of it from `react-devtools-core/standalone` and `react-devtools-inline`.

### CODE REVIEW

File: packages/react-devtools-shared/src/devtools/views/Settings/SettingsModalContext.js [hunk 1]
```diff
@@ -24,7 +24,6 @@ import type {FrontendBridge} from '../../../bridge';
 import type {DevToolsHookSettings} from '../../../backend/types';
 import type Store from '../../store';
 
-export type DisplayDensity = 'comfortable' | 'compact';
```
Comment (hoxyq) [inline packages/react-devtools-shared/src/devtools/views/Settings/SettingsModalContext.js:27]: Unused as well, just a drive-by cleanup.

**Review by vzaidman — APPROVED** [review]
